DAQ_System/Source/main.cpp

```diff
@@ -58,48 +67,72 @@ int main() {
 
     printf("--- CSV File Test ---\n");
 
-    // mounting the file system has moved to the constructor of the data logger class
-
+    // used for opening a file
     char file_name[16] = "\0";
     uint8_t file_name_status = 0;
-
     char write_buffer[BUFFER_SIZE] = "\0";
-
-    for(int i = 0; i < 1000; i++) {
-        // Increment file_name
-        snprintf(file_name, sizeof(file_name), "/fs/data%d.csv", i);
-        file_name_status = data_logger->FileOpen(file_name);
-
-        if (file_name_status == 2) {
-            // File not found, found a unique name
-            printf("Opened %s\n", file_name);
-            break;
-        } else {
-            // File already exists
-            data_logger->FileClose();
+    
+    // use internal pullup for pushbutton 
+    pb.mode(PullUp);
+    // delay for initial pullup to take effect
+    wait_ns(100);
+    // attach the address of the interrupt handler routine for pushbutton
+    pb.fall(&pb_hit_interrupt);
+
+    uint8_t open_file = 0;
+
+    // super loop
+    printf("Entering Super Loop\n");
+    while(true) {
+        wait_ns(100);
+        // check data logger flag
+        if(data_logger_flag && !open_file) {
```

Comment:
I like this open file check. Maybe this can be moved to the data logging class?

Reply:
We will come back to this one. Perhaps this is a good use case of data sharing using a pointer.